### PR TITLE
feat(CosmosPackagesStore): implement service/describe in store

### PR DESCRIPTION
**Pull request by ahoskins**
~⚠️  *Depends on #2404 (provides the actions)*~

**Epic**: Store support for [Edit Configuration Settings](https://jira.mesosphere.com/browse/DCOS-17721).

**Summary**: Adds support to cosmos store for `service/describe` endpoint.

**Quick description**: expose a new method `.fetchServiceDescription()` on the cosmos packages store. A successful result of this will be just like `.fetchPackageDescription()` - it will set the `packageDetails` field in the store. The component (not built yet) can then use this the populate its form.
  
Closes DCOS-18200. 

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

File: src/js/stores/CosmosPackagesStore.js [hunk 1]
```diff
@@ -78,7 +82,10 @@ class CosmosPackagesStore extends GetSetBaseStore {
         repositoryAddSuccess: COSMOS_REPOSITORY_ADD_SUCCESS,
         repositoryAddError: COSMOS_REPOSITORY_ADD_ERROR,
         repositoryDeleteSuccess: COSMOS_REPOSITORY_DELETE_SUCCESS,
-        repositoryDeleteError: COSMOS_REPOSITORY_DELETE_ERROR
+        repositoryDeleteError: COSMOS_REPOSITORY_DELETE_ERROR,
+
+        cosmosServiceDescriptionSuccess: COSMOS_SERVICE_DESCRIBE_CHANGE,
```
Comment (ahoskins) [inline src/js/stores/CosmosPackagesStore.js]: I will rename this. With this name, the StoreMixin names the callback `onCosmosPackagesStoreCosmosServiceDescriptionSuccess()`...which is incredibly wordy.

**Comment by ahoskins** [comment]
Alfred please test this

File: src/js/stores/CosmosPackagesStore.js [hunk 2]
```diff
@@ -87,7 +91,10 @@ class CosmosPackagesStore extends GetSetBaseStore {
         repositoryAddSuccess: COSMOS_REPOSITORY_ADD_SUCCESS,
         repositoryAddError: COSMOS_REPOSITORY_ADD_ERROR,
         repositoryDeleteSuccess: COSMOS_REPOSITORY_DELETE_SUCCESS,
-        repositoryDeleteError: COSMOS_REPOSITORY_DELETE_ERROR
+        repositoryDeleteError: COSMOS_REPOSITORY_DELETE_ERROR,
+
+        serviceDescriptionSuccess: COSMOS_SERVICE_DESCRIBE_CHANGE,
```
Comment (weblancaster) [inline src/js/stores/CosmosPackagesStore.js]: not big deal but based on the other constants shouldn't this be `COSMOS_SERVICE_DESCRIBE_SUCCESS`?

File: src/js/stores/CosmosPackagesStore.js [hunk 3]
```diff
@@ -355,6 +372,18 @@ class CosmosPackagesStore extends GetSetBaseStore {
     this.emit(COSMOS_REPOSITORIES_SUCCESS);
   }
 
+  processServiceDescriptionSuccess(cosmosPackage, name) {
```
Comment (weblancaster) [inline src/js/stores/CosmosPackagesStore.js]: Why are we using this method to do the same thing as `processPackageDescriptionSuccess`?

File: src/js/stores/CosmosPackagesStore.js [hunk 4]
```diff
@@ -355,6 +372,18 @@ class CosmosPackagesStore extends GetSetBaseStore {
...
+    this.set({ packageDetails: cosmosPackage });
+
+    this.emit(COSMOS_SERVICE_DESCRIBE_CHANGE, name);
+  }
+
+  processServiceDescriptionError(error, name) {
```
Comment (weblancaster) [inline src/js/stores/CosmosPackagesStore.js]: seems like you have this method `processPackageDescriptionError `

File: src/js/stores/CosmosPackagesStore.js (hunk 3 above)
Comment (ahoskins) [reply src/js/stores/CosmosPackagesStore.js]: The only different is the emitted event. I think they should be kept using their own separate methods.

File: src/js/stores/CosmosPackagesStore.js (hunk 4 above)
Comment (ahoskins) [reply src/js/stores/CosmosPackagesStore.js]: Same as above. The emitted event is different.

File: src/js/stores/CosmosPackagesStore.js (hunk 2 above)
Comment (ahoskins) [reply src/js/stores/CosmosPackagesStore.js]: `COSMOS_SERVICE_DESCRIBE_CHANGE` follows the same naming as the existing `COSMOS_DESCRIBE_CHANGE`. These are "sister" endpoints, so I think current naming makes sense.

Comment (ahoskins) [reply src/js/stores/CosmosPackagesStore.js]: Judging from this file, we seem to use "success" when an action was performed that affects the cluster, and we use "change" when fetching some kind of state.

File: src/js/stores/CosmosPackagesStore.js (hunk 3 above)
Comment (ahoskins) [reply src/js/stores/CosmosPackagesStore.js]: Talked offline and decided we will introduce a new field `serviceDetails` to make this nicer.

File: src/js/components/modals/InstallPackageModal.js [hunk 5]
```diff
@@ -61,7 +61,7 @@ class InstallPackageModal
         name: "cosmosPackages",
         events: [
           "descriptionError",
```
Comment (MatApple) [inline src/js/components/modals/InstallPackageModal.js]: We need to change the `descriptionError` too don't we?

Comment (ahoskins) [reply src/js/components/modals/InstallPackageModal.js]: Yes, good catch

**Comment by ahoskins** [comment]
@MatApple I only grepped for `descriptionSuccess` and `descriptionError` within `dcos-ui` and `dcos-ui-plugins-private`. There is no other repo that would use these events, right?

**Comment by MatApple** [comment]
@ahoskins no other repos 👍

**Comment by MatApple** [comment]
💯

**Pull request merged**